src/Symfony/Contracts/Cache/CacheContractsTrait.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Contracts\Cache;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Cache\InvalidArgumentException;
+
+/**
+ * An implementation of CacheInterface for PSR-6 CacheItemPoolInterface classes.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+trait CacheContractsTrait
```

Comment:
The responsibility is mostly given by the interface this allows implementing.
The specific added responsibility here is providing an implementation of get+delete using PSR-6 methods, so that implementing the contracts interface is a oneliner.
Something like `CacheInterfaceImplementationForCacheItemPoolInterfaceTrait` would be more precise, but I fear that's not really better. At least, from a consumer pov (ie a PSR-6 implementer), having `use CacheContractsTrait;` might be a good enough hint to quickly remind one about what this is doing.
But I'd happily listen to better suggestions.

> opportunity to improve name of CacheIterface as well

Sure, naming is hard :) Any suggestions? If we really want an alternative name, my proposal would be `CacheStorageInterface`. This has the benefit of not colliding with PSR-16 `CacheInterface` (if we should care.)

Reply:
Yep I like CacheStorage* :)

Isn't tight coupling of this trait to interface sign it would be better to replace these with abstract class?

Reply:
`CacheInterface` is simpler. Im not sure how many times I've been fighting with PSR-6 interfaces' complicated names. 
I vote for no rename of the interface. 
The trait however. Not sure... Dont care much =)